### PR TITLE
fix(RHINENG-9037) Return tags with null values & refactor

### PR DESCRIPTION
**Pull request by kruai**
# Overview

This PR is being created to address [RHINENG-9037](https://issues.redhat.com/browse/RHINENG-9037).
It fixes a bug which was preventing tags with null values to be returned by API endpoints. It also refactors the relevant code a bit, and adds a unit test.

## PR Checklist

- [x] Keep PR title short, ideally under 72 characters
- [x] Descriptive comments provided in complex code blocks
- [x] Tests: validate optimal/expected output
- [x] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [x] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: api/host_query_db.py [hunk 1]
```diff
@@ -194,6 +194,23 @@ def get_host_tags_list_by_id_list(
     return host_tags_dict, len(host_id_list)
 
 
+def _add_tag_values(host_namespace, tag_key, tag_value, host_id, host_tags, host_tags_tracker) -> Tuple[dict, dict]:
+    host_tag_str = (
+        f"{_convert_null_string(host_namespace)}/{_convert_null_string(tag_key)}={_convert_null_string(tag_value)}"
+    )
```
Comment (chambridge) [inline api/host_query_db.py:200]:
```suggestion
    host_tag_str = (
        f"{_convert_null_string(host_namespace)}/{_convert_null_string(tag_key)}={_convert_null_string(tag_value)}"
    )
    if not _convert_null_string(tag_value):
        host_tag_str = (
            f"{_convert_null_string(host_namespace)}/{_convert_null_string(tag_key)}="
        )
```

If the value here is a null or [] then the `host_tag_str` would end with `=None` which wouldn't be good for the regex search.

**Review by chambridge — COMMENTED** [review]
One minor comment on the refactor.

File: api/host_query_db.py (hunk 1 above)
Comment (kruai) [reply api/host_query_db.py:200]: Oops, brainfart. For some reason that equated to blank. Fixing

Comment (kruai) [reply api/host_query_db.py]: OK, fixed. I just modified the value used in the `host_tag_str`

**Pull request merged**